lib/rubygems/core_ext/kernel_require.rb

```diff
@@ -95,7 +95,7 @@ def require(path) # :doc:
       if found_specs.empty?
         found_specs = Gem::Specification.find_in_unresolved_tree path
 
-        found_specs.each(&:activate)
+        found_specs.each(&:activate) unless default_spec
```

Comment:
Can we skip line 96 if `default_spec` is present then?

Reply:
Good catch!

Reply:
I pushed a simpler patch. Once we activate the default gem, we can skip the rest of the stuff!